ReactAndroid/src/main/java/com/facebook/react/modules/statusbar/StatusBarModule.java

```diff
@@ -183,23 +184,37 @@ public void setStyle(@Nullable final String style) {
           "StatusBarModule: Ignored status bar change, current activity is null.");
       return;
     }
-
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-      UiThreadUtil.runOnUiThread(
-          new Runnable() {
-            @TargetApi(Build.VERSION_CODES.M)
-            @Override
-            public void run() {
-              View decorView = activity.getWindow().getDecorView();
-              int systemUiVisibilityFlags = decorView.getSystemUiVisibility();
-              if ("dark-content".equals(style)) {
-                systemUiVisibilityFlags |= View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR;
-              } else {
-                systemUiVisibilityFlags &= ~View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR;
-              }
-              decorView.setSystemUiVisibility(systemUiVisibilityFlags);
+    
+    UiThreadUtil.runOnUiThread(
+      new Runnable() {
+        @TargetApi(Build.VERSION_CODES.R)
+        @Override
+        public void run() {
+          if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            WindowInsetsController insetsController = activity.getWindow().getInsetsController();
+            if ("light-content".equals(style)) {
+              // light-content means white icons on a dark status bar
+              insetsController.setSystemBarsAppearance(
+                0,
+                WindowInsetsController.APPEARANCE_LIGHT_STATUS_BARS
+              );
+            } else {
+              insetsController.setSystemBarsAppearance(
```

Comment:
@ieatfood I just realized that this would be a breaking behavior comparing to the `systemUiVisibility` approach.

Before, we only set `LIGHT_STATUS_BAR` for `"dark-content"`, while now we set it for both `"default"` and `"dark-content"`. Although it could make more sense to default to `"dark-content"` within Android, could you use the old logic to make sure we don't break people apps? (or introduce disparity between versions 🙂 )

Reply:
You're right, I misread the original logic and though I had to flip it. Updated.